base/lttagging.dtx

```diff
@@ -104,49 +104,55 @@
 %  debugging strings\ldots
 %
 %
+% \DescribeMacro\UseExpandableTaggingSocket
 % \DescribeMacro\UseTaggingSocket
+% \DescribeMacro\tag_socket_use_expandable:n
 % \DescribeMacro\tag_socket_use:n
 % \DescribeMacro\tag_socket_use:nn
-%  To support tagging in packages we use sockets with names starting
-%  with \texttt{tagsupport/}. Usually, these sockets have exactly two
-%  plugs defined:
-%  \plug{noop} (when no tagging is requested or tagging is not wanted
-%  for some reason) and a second plug that enables the tagging. There
-%  may be more, e.g., tagging with special debugging, etc., but right
-%  now it is usually just on or off.
-%
-%  Given that we sometimes have to suspend tagging, it would be fairly
-%  inefficient to put different plugs into these sockets whenever that
-%  happens. We therefore offer \cs{UseTaggingSocket} which is like
-%  \cs{UseSocket} except that the socket name is specified without
-%  \texttt{tagsupport/}, i.e.,
+% \DescribeMacro\tag_socket_use:nnn
+% Given that we sometimes have to suspend tagging, it would be fairly
+% inefficient to put different plugs into these sockets whenever that
+% happens. We therefore offer \cs{UseTaggingSocket} which is like
+% \cs{UseSocket} except that is expects a socket starting with
+% \texttt{tagsupport/} but the socket name is specified without
+% this prefix, i.e.,
 % \begin{quote}
 %   \verb=\UseTaggingSocket{foo}=      $\to$
 %   \verb=\UseSocket{tagsupport/foo}=
-% \end{quote}
-%  Beside being slightly shorter, the big advantage is that this way
-%  we can change \cs{UseTaggingSocket} to do nothing when tagging is
-%  suspended with \cs{SuspendTagging} instead of changing the plugs of
-%  the tagging support sockets back and forth.
+% \end{quote}.
 %
-%  This means that with \cs{SuspendTagging} in force all socket
-%  arguments are ignored (including the case when there is a single
-%  one), i.e., all sockets used with \cs{UseTaggingSocket} behave as
-%  if the \text{noop} plug is assigned.
+%  Beside being slightly shorter, the big advantage is that this way
+%  we can change \cs{UseTaggingSocket} to do nothing by switching a boolean
+%  instead of changing the plugs of the tagging support sockets back and forth.
+%
+% Usually, these sockets have exactly two plugs defined.
+% When using \cs{UseTaggingSocket} the first plug is always
+% used if tagging is suspended.
+% For the sockets with zero and one argument
+% this is the \texttt{noop} plug, so they gobble the
+% argument in such a case. The assigned plug use when tagging is enable.
```

Comment:
sentence at the end of the line is no sentence :-) but again I think the whole para need rewriting